### ISSUE TITLE
Detecting yet-to-be-loaded models

### ISSUE THREAD

**Issue by chbonser** [orbitjs/ember-orbit#353]
What is the recommended way of detecting whether a model has been loaded yet?

Prior to https://github.com/orbitjs/ember-orbit/pull/350 (0.17.0-beta.13) I was checking the presence of an `updatedAt` attr on each model. Now attempting to do so throws an error.

Can the Model class provide an `isLoaded` or similar method? I perused https://github.com/orbitjs/ember-orbit/blob/main/addon/-private/model.ts and didn't see something already existing.

**Comment by dgeb** [comment]
@chbonser Sorry for the breakage that #350 introduced. I'm rethinking mutations via models a bit and need to understand your use case that involves a model instance with no backing data (not even an identity) in its cache. Can you sketch it out with a short example?

**Comment by chbonser** [comment]
No worries on the breaking change. It is a beta release after all.

### Example
Here is my use case (roughly):
Suppose you have a `Universe` model which has many `Star`. When you fetch a `Universe` the API responds like this:
```
{
  data: {
    attributes: {...}
    relationships: {
      stars: { data: [{id: '1', type: 'moon'}, ...] }
    }
  ...
}
```

Including `stars` in that request is too expensive so we don't do it. 

What do we know now though? 
1. We know how many stars there are in the universe and their identities
2. We know how many are loaded in the store

We can now query for batches of stars by pulling the ids from the identities and sending the array of ids as a filter on our request.

Before #350 I could call `universe.stars.filter(star => !star.updatedAt)` to get a list of unloaded starts. After #350 I can achieve the same ends but would need to catch the exception that is raised instead.

### Alternate Solutions
I recognize that there are alternate ways I could have solved the "has many relationships are expensive to load" problem but the ones that come to mind do have drawbacks. 

For example, I could have left the stars relationship off of my universe payload and included a count of stars instead. Then instead of requesting batches of stars using a filter, I could use a more classic paginated index. This however does not account for stars being created/destroyed. It is not just the count but the identities of the stars that matter.

**Comment by dgeb** [comment]
Thanks for clarifying, @chbonser. This seems like a reasonable use case, and certainly one I overlooked. I'll make sure you have a workaround soon, even if it means just undoing part of #350.

**Comment by dgeb** [comment]
Resolved in v0.17.0-beta.14
